### ISSUE TITLE
Refactor Deferred and Async HOWTOS

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7255]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)| @puzzlement reported|
|-|-|
|Trac ID|trac#504|
|Type|enhancement|
|Created|2004-02-15 08:11:22Z|

<details><summary>Searchable metadata</summary>

```
trac-id__504 504
type__enhancement enhancement
reporter__hypatia hypatia
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__documentation_core documentation core
time__1076832682000000 1076832682000000
changetime__1474221881907648 1474221881907648
version__ 
owner__ 
cc__spiv cc__hypatia
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
The Deferred HOWTO currently contains a lot of information which doesn't flow
naturally. It might be better to merge some of it into the async howto, or
perhaps to create an entire async chapter.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
OK, working on this again.

Some notes on the async howto:

1. There's a fair bit of handwaving in the advantages and disadvantages section
that needs to be properly explained. (To me, so I can explain it to other people.)

2. There's probably room for an entire new document on design of Python programs
to fit an asynch framework. The latter half of this document should probably
move there, and this one can start to discuss callbacks and so on: the
fundamentals of async are sort of missing.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
Random suggestion from #twisted

&lt;robertg> You might want to illustrate deferreds better by showing that one can
do other things (a simple print statement might work) while waiting for a callback.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Ok, there are now three separate documents. They're not great, but I judge the
refactoring at least to be complete.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc commented|
|-|-|

In changeset ca266e967a728f1d0f2f4aebeb966b4ad7af1038
```
#!CommitTicketReference repository="" revision="ca266e967a728f1d0f2f4aebeb966b4ad7af1038"
Merge pull request #504 from twisted/8765-rodrigc-trial-removal

Author: rodrigc
Reviewer: adiroiban
Fixes: #8765

Remove bin/trial from the tree
```
